server/http_handler.go

```diff
@@ -909,6 +915,128 @@ func (h flashReplicaHandler) handleStatusReport(w http.ResponseWriter, req *http
 		zap.Error(err))
 }
 
+type schemaTableStorage struct {
+	TableSchema   string `json:"table_schema"`
+	TableName     string `json:"table_name"`
+	TableRows     int64  `json:"table_rows"`
+	AvgRowLength  int64  `json:"avg_row_length"`
+	DataLength    int64  `json:"data_length"`
+	MaxDataLength int64  `json:"max_data_length"`
+	IndexLength   int64  `json:"index_length"`
+	DataFree      int64  `json:"data_free"`
+}
+
+func getSchemaTablesStorageInfo(h *schemaStorageHandler, schema *model.CIStr, table *model.CIStr) (messages []*schemaTableStorage, err error) {
+	var s session.Session
+	if s, err = session.CreateSession(h.Store); err != nil {
+		return
+	}
+	if s != nil {
```

Comment:
it seems we do not need to do anything when s is nil, could we process this situation before, so we needn't always remember this situation until the end of the function.

Reply:
done